src/Tizen.NUI/src/public/Text/Spans/ColorSpanBuilder.cs

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright(c) 2022 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+using System;
+using System.ComponentModel;
+
+namespace Tizen.NUI.Text.Spans
+{
+    /// <summary>
+    /// </summary>
+    // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ColorSpanBuilder : Disposable
+    {
+
+        /// <summary>
+        /// Creates a new ColorSpanBuilder object<br />
+        /// </summary>
+        // This will be public opened after ACR done. (Before ACR, need to be hidden as Inhouse API)
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static ColorSpanBuilder Initialize( )
```

Comment:
The builder pattern is a good pattern when they have many configurable fields. 
But, ColorSpan has only one filed

So, it not good for builder pattern
I think it just enough factory method
```suggestion
        public static ColorSpan WithForegroundColor(Color color)
```

Reply:
on android also builder pattern was not used to create span object
The builder pattern was only used to create spanableStrings, not spans.

Reply:
The factory method is added according to your suggestion. I agree with you this is easier for developer.

But, we need to keep Builder because we are adding more spans and most of them have many attributes.
We need to provide the same experience on all spans.
Also, we will add more APIs in builder to allow developer construct ColorSpan with other options. 

Thanks for your time and feedback.

Reply:
Could you give me one example that has many attribute?

Reply:
I think We can split a complex span into individual spans, for example FontSpan could be split into FontSizeSpan, FontFaceSpan, FontStyleSpan, FontColorSpan.

anyway, I think a complex span is not normal in span so Builder pattern should not apply to a common pattern on span

Reply:
Here's a PR for FontSpan. But it is under testing.
https://github.com/shrouqsabah/TizenFX/pull/3

We can not split complex span into simple spans:
1) We have to integrate spans with tags in markup. Each tag should be mapped to span. ( we have features depend on this integration)
2) If we split complex spans then the developer will create maybe 3 to 5 spans for same range(s), imagine the number of spans and number of calling for attach. (we want to create a general span like the one in markup)
3) We can help developer through builders to create spans in many different options (especially we are using fluent builder)
 
Thanks for your time and feedback

Reply:
OK, I respect you